src/VisualStudio/Core/Def/Implementation/ProjectSystem/IRunningDocumentTableEventListener.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem
+{
+    /// <summary>
+    /// Defines the methods that get called by the <see cref="RunningDocumentTableEventTracker"/>
+    /// for getting notified about running document table events.
+    /// </summary>
+    internal interface IRunningDocumentTableEventListener
+    {
+        void OnCloseDocument(uint docCookie, string moniker);
+
+        void OnRefreshDocumentContext(uint docCookie, string moniker);
+
+        void OnReloadDocumentData(uint docCookie, string moniker);
+
+        void OnBeforeOpenDocument(uint docCookie, string moniker, ITextBuffer textBuffer);
+
+        void OnInitializedDocument(uint docCookie, string moniker, ITextBuffer textBuffer);
```

Comment:
What's the difference between this one and the one above? I would assume we'd be abstracting the difference here -- I can't imagine a user of this that doesn't want to treat both equally.

Reply:
For the open file tracker they do the exact same thing.  But for the misc workspace, it only does the DocumentInitialized version.  I guessed there was a reason that it was done differently, do you know if that's correct?